# Viral Genomics and Bioinformatics Asia 2022
Base for the Viral Genomics and Bioinformatics Repository

## Course overview
We are pleased to announce the next Viral Genomics and Bioinformatics in collaboration with COVID-19 Genomics (COG) Train (COG-Train).

Viruses are a huge public health burden all around the world. In Asia, viral infections such as influenza, dengue fever, hand, foot and mouth disease, chikungunya, and viral hepatitis kill hundreds of thousands of people each year.  Apart from health issues, emerging viral illnesses and epidemics have a global economic impact. According to the World Bank, the SARS-Cov-2 pandemic caused a more than $10 trillion deficit in the global economy in 2020-2021 alone. This demonstrates the impact of viral illnesses not only on human and animal health, but also on social and economic growth of the world.

Technological advances in science and research are making it possible to rapid detection, surveillance and management of viral diseases. In recent years, next generation sequencing (NGS) technologies have played an important role in the identification and classification of viruses, in the detection of drug resistance mutations and in the treatment and surveillance of viral diseases. Early identification of a virus and quick analysis of its genome will aid towards better treatment and help in controlling the disease spread. Along with the advances in the sequencing technologies, it is necessary to develop data management methods, computational analysis tools and human expertise to cope with the sequence deluge. The number of researchers able to interpret viral high throughput sequence data is currently limited.   

The Viral Genomics and Bioinformatics course will teach participants how to analyse  viral genome sequences and interpret genomic data obtained by high-throughput sequencing. The use of the command line to analyse viral NGS data, quality control of sequences, reference mapping and de novo assembly of viral genomes, pathogen identification from metagenomics data, and creating phylogenetic trees will be covered in the course. 

[Course website](https://coursesandconferences.wellcomeconnectingscience.org/event/viral-genomics-and-bioinformatics-asia-20220822/)

## Instructors
- [Sreenu Vattipally](https://www.gla.ac.uk/researchinstitutes/iii/staff/sreenuvattipally/), MRC-University of Glasgow Centre for Virus Research, UK
- [Urmila Kulkarni Kale](https://www.researchgate.net/profile/Urmila_Kulkarni-Kale/), Savitribai Phule Pune University
- [Le Van Tan](), Oxford University Clinical Research Unit (OUCRU)
- [Srikeerthana Kuchi](), MRC-University of Glasgow Centre for Virus Research, UK
- [Richard Orton](https://www.gla.ac.uk/researchinstitutes/iii/staff/richardorton/), MRC-University of Glasgow Centre for Virus Research, UK
- [To Nguyen](), Oxford University Clinical Research Unit (OUCRU)
- [October Sessions](https://sph.nus.edu.sg/faculty-directory/sessions-october-michael/), National University of Singapore
- [Sunitha Manjari Kasibhatla](), Savitribai Phule Pune University
- [Dhiraj Dhotre](), National Center for Cell Sciences, Pune 
- [David Studholme](https://biosciences.exeter.ac.uk/staff/profile/index.php?web_id=david_studholme), University of Exeter
- [Kwanrutai Mairang](), Faculty of Medicine Siriraj Hospital, Mahidol University



## Timetable
### Overview


### Detailed timetable
[Download here]()


## Course manual
**Module 1 - Introduction to Linux** 

**Module 2 - Linux text processing**  

**Module 3 - Introduction to NGS sequencing**  
[Online Manual - Introduction to NGS](Modules/Introduction_to_NGS.md)

  
**Module 4 - NGS file formats and data & QC**  
[Online Manual - NGS data Quality Control QC](Modules/NGS_file_formats_and_QC.md)

**Module 5 - Introduction to BASH scripting**  
[PDF Manual - Bash Scripting](Modules/BASH_scripting.pdf)

**Module 6 - Reference Alignment**  
[Online Manual - Reference Alignment](Modules/Reference_alignment.md)
 
**Module 7 - Coverage Plots and Stats**  
[Online Manual - Coverage Plots and stats](Modules/Coverage_plots_and_statistics.md)
 
**Module 8 - Consensus & variant calling**  
[Online Manual - Consensus & variant calling](Modules/Consensus-and-variant-calling-theory-and-practical.md)

**Module 9 - De Novo assembly**  

**Module 10 - Pathogen sequence detection using metagenomics**  
[Online Manual - Metagenomics](course_data/Pathogen_sequence_detection_using_metagenomics/Module10-Metagnomics.md)

[PDF Manual - Metagenomics](course_data/Pathogen_sequence_detection_using_metagenomics/Module10-Metagnomics.pdf)


**Module 11 - Introduction to phylogenetic methods**  

[Online Manual - Phylogeny and Virus Typing](Modules/Phylogeny_&_Virus_Typing_HandsOnEx.md)

**Module 12 - SARS-CoV-2 workflows**  

  
**Task 1 - **  

**Task 2 - **  

**Appendix**  




Any reuse of the course materials, data or code is encouraged with due acknowledgement.

******
## License
<a rel="license" href="http://creativecommons.org/licenses/by/4.0/"><img alt="Creative Commons Licence" style="border-width:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" /></a><br />This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.

